pkg/probe/exec/exec.go

```diff
@@ -17,39 +17,102 @@ limitations under the License.
 package exec
 
 import (
+	"bytes"
+	"io"
+
+	"k8s.io/klog"
 	"k8s.io/kubernetes/pkg/probe"
 	"k8s.io/utils/exec"
+)
 
-	"k8s.io/klog"
+const (
+	maxReadBytes = 2 * 1024 // 2kb
 )
 
 // New creates a Prober.
 func New() Prober {
-	return execProber{}
+	return execProber{
+		maxReadBytes: maxReadBytes,
+	}
 }
 
 // Prober is an interface defining the Probe object for container readiness/liveness checks.
 type Prober interface {
 	Probe(e exec.Cmd) (probe.Result, string, error)
 }
 
-type execProber struct{}
+type execProber struct {
+	maxReadBytes int64
+}
 
 // Probe executes a command to check the liveness/readiness of container
 // from executing a command. Returns the Result status, command output, and
 // errors if any.
 func (pr execProber) Probe(e exec.Cmd) (probe.Result, string, error) {
```

Comment:
note: This really needs a timeout... Not something for this PR though.

Reply:
Oh, looks like https://github.com/kubernetes/kubernetes/pull/58925 is tackling this.

Reply:
Yep, I think we want #58952 in any case, regardless of this PR.

I think both (limiting the runtime of a probe *and* limiting the amount of mem the probe wastes on stdout & stderr) are valuable, the former may even more valuable than the later.